helpers/helpers_emba_prepare.sh

```diff
@@ -489,6 +490,18 @@ detect_root_dir_helper() {
     fi
   done
 
+  mapfile -t ROOTx_PATH < <(find "$SEARCH_PATH" -xdev -path "*bin/sh" | sed -E 's/\/.?bin\/sh//')
+  for R_PATH in "${ROOTx_PATH[@]}"; do
+    if [[ -d "$R_PATH" ]]; then
+      ROOT_PATH+=( "$R_PATH" )
+      if [[ -n "$MECHANISM" ]] && ! echo "$MECHANISM" | grep -q "file names"; then
+        MECHANISM="$MECHANISM / file names"
```

Comment:
let's change this from _file names_ to something like _shells_. Could you please also change this in bash detection?
I think we can improve this detection a bit if we are going to pimp the find command the following way:
```
find "$SEARCH_PATH" -xdev -path "*bin/sh" -exec file {} \; | grep "ELF" | cut -d: -f1 | sed -E 's/\/.?bin\/sh//'
```
Same for the bash block.

Reply:
I have added this, I also reversed the logic of settings the MECHANISM, still have to test if it works correctly. I also want to do more tests, because it feels odd that my filesystem was not detected as root filesystem when looking at the other checks.